### DISCUSSION TITLE
Disable the preview of shader materials in the 2D window

### DISCUSSION THREAD

**Discussion by TweedleDim** [godotengine/godot-proposals#10198] [Editor]
Same request as : https://github.com/godotengine/godot/issues/21361

I would like to be able to not display the effects of shaders in the editor.
An option to toggle in the View menu of the 2D Screen would seem to be the right place.
Or in the Editor Settings ?

Thumbs up if you think it could be a useful feature !

**Comment by Calinou** [comment]
Out of curiosity, why are you looking to disable shader previews in the editor?

**Comment by TweedleDim** [comment]
Well it's purely for convinience. In the present case it's because i have a background i animate through a shader, just a simple slow sliding effect. But it flickers a lot and it bothers me to see this thing moving while i'm working on my level.

**Comment by riverfr0zen** [comment]
In my case, running Wayland with COSMIC DE, the shader preview in the 2D window is significantly slowing down my computer and making it stutter and lag. Godot becomes fairly unusable in editing the scene, especially if it's the main scene. If I switch to a different scene without any shaders, the performance issue stops. 

The shaders in question are not particularly complex, and they run fine in the ShaderEditor. It's only when working in the 2D window of a scene that displays the shaders that performance goes down.

Would really appreciate a way to disable it--for me this isn't just a convenience issue now.

**Reply by Calinou** [reply]
This PR will add an option to address the performance issue while keeping shaders intact (although they won't animate anymore):

- https://github.com/godotengine/godot/pull/113224
